### PR TITLE
Mark mleap as deprecated 

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Mark mleap as a deprecated flavor

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

mleap is being marked for deprecation. Please use ONNX as a replacement.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/mleap.py [hunk 1]
```diff
@@ -2,6 +2,10 @@
 The ``mlflow.mleap`` module provides an API for saving Spark MLLib models using the
 `MLeap <https://github.com/combust/mleap>`_ persistence mechanism.
 
+WARNING:
```
Comment (harupy) [inline mlflow/mleap.py:5]: Is this a valid rst syntax?

File: mlflow/mleap.py [hunk 2]
```diff
@@ -21,13 +25,14 @@
...
 from mlflow.utils.file_utils import path_to_local_file_uri
-from mlflow.utils.annotations import keyword_only
+from mlflow.utils.annotations import keyword_only, deprecated
 
 FLAVOR_NAME = "mleap"
 
 _logger = logging.getLogger(__name__)
 
 
+@deprecated(alternative="mlflow.onnx", since="2.6.0")
```
Comment (harupy) [inline mlflow/mleap.py:35]: Is `mlflow.onnx` alternative?

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @BenWilson2 !

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/mleap.py (hunk 2 above)
Comment (WeichenXu123) [reply mlflow/mleap.py:35]: Same question, I think onnx cannot directly support exporting spark model.

File: mlflow/mleap.py (hunk 1 above)
Comment (WeichenXu123) [reply mlflow/mleap.py:5]:
```suggestion
.. warning::
```

File: mlflow/mleap.py [hunk 3]
```diff
@@ -2,6 +2,10 @@
 The ``mlflow.mleap`` module provides an API for saving Spark MLLib models using the
 `MLeap <https://github.com/combust/mleap>`_ persistence mechanism.
 
+WARNING:
+
+    The mleap flavor is deprecated and will be removed in a future release of MLflow.
+
 NOTE:
```
Comment (WeichenXu123) [inline mlflow/mleap.py:9]: .. note::

File: mlflow/mleap.py [hunk 4]
```diff
@@ -2,6 +2,10 @@
 The ``mlflow.mleap`` module provides an API for saving Spark MLLib models using the
 `MLeap <https://github.com/combust/mleap>`_ persistence mechanism.
```
Comment (WeichenXu123) [inline mlflow/mleap.py:3]: Btw, The new added spark ml connect component also achieves the goal that run model inference locally without spark context, but note that it does not support legacy spark ML models.

Comment (BenWilson2) [reply mlflow/mleap.py:3]: I think we should definitely add in the support once the Spark3.5 changes are ready for default-on (hopefully this quarter :) )

File: mlflow/mleap.py (hunk 2 above)
Comment (indranilr) [reply mlflow/mleap.py:35]: Any change to this ? How does one export Spark Xgboost model to ONNX ?

**Pull request merged**